### PR TITLE
Build pytorch 1.13.0

### PR DESCRIPTION
**Pull request by mikemhenry**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by mikemhenry** [comment]
@conda-forge-admin, please rerender

**Comment by mikemhenry** [comment]
So far it looks like the failures are from old python versions and new pytorch versions, which is to be expected since the newest version of pytorch doesn't support the oldest version of python we are trying to support.

**Comment by peastman** [comment]
That makes sense.  Is there a way to exclude that combination?  I'd be hesitant to drop Python 3.8 support altogether.  It's not that old, and it still has almost two years left before its end of support date.

**Comment by mikemhenry** [comment]
Looking at the failed builds, the only issue is that for pytorch 1.13, we need cuda 11.2 or newer. I can try and remove the builds, or we can just merge

**Comment by peastman** [comment]
Let's remove those builds.  Merging while there are builds failing isn't generally a good idea.

**Comment by raimis** [comment]
I agree! CUDA 12 is out, so it is time to move away from CUDA 10.

**Comment by peastman** [comment]
I didn't mean to drop 10.2.  There are still a lot of people using it.  I just meant to remove the particular combinations that aren't supported.

**Comment by mikemhenry** [comment]
Sounds good! I am working on getting the selectors working just right...

**Comment by h-vetinari** [comment]
> I didn't mean to drop 10.2. There are still a lot of people using it. I just meant to remove the particular combinations that aren't supported.

conda-forge does not build pytorch>=1.13 for CUDA<1.12 anymore, see  https://github.com/conda-forge/conda-forge-pinning-feedstock/issues/3491. Unless someone steps up to _consistently_ debug and build those variants on the pytorch feedstock, you have no choice but to drop old CUDA for newer pytorch.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ source:
   git_rev: "994f92fde4ce710a4db5557b6bc22756d5f9d2b1"
 
 build:
-  number: 0
+  number: 2
   string: cuda{{ cuda_compiler_version | replace('.', '') }}py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [cuda_compiler_version != "None"]
   string: cpu_py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}                                      # [cuda_compiler_version == "None"]
   skip: true  # [win or (linux and cuda_compiler_version in (undefined, 'None', '10.2'))]
```
Comment (h-vetinari) [inline recipe/meta.yaml:19]:
```suggestion
  skip: true  # [win or (linux and cuda_compiler_version in (undefined, 'None', '10.2'))]
  # as of pytorch 1.13, conda-forge only builds for CUDA 11.2+, see
  # https://github.com/conda-forge/conda-forge-pinning-feedstock/issues/3491
  skip: true  # [pytorch == "1.13" and cuda_compiler_version in ("10.2", "11.0", "11.1")]
```

**Review by h-vetinari — COMMENTED** [review]
> Sounds good! I am working on getting the selectors working just right...

Here's a suggestion that should work

File: recipe/meta.yaml (hunk 1 above)
Comment (peastman) [reply recipe/meta.yaml:19]: That looks right to me.

**Comment by peastman** [comment]
@conda-forge-admin, please rerender

**Comment by peastman** [comment]
We added the line

```yaml
skip: true  # [pytorch == "1.13" and cuda_compiler_version in ("10.2", "11.0", "11.1")]
```

but it still tried to build with pytorch 1.13 and old CUDA versions.  I first assumed that was because the pytorch version was actually 1.13.0, not 1.13, so I changed the comparison to `pytorch >= "1.13"`.  But now it fails with the error

> '>=' not supported between instances of 'bool' and 'str'

which suggests that `pytorch` is a bool, not a string?  What's the correct way to check the pytorch version?

File: .ci_support/linux_64_c_compiler_version10cuda_compiler_version11.1cxx_compiler_version10python3.10.____cpythonpytorch1.13.0.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+docker_image:
+- quay.io/condaforge/linux-anvil-cuda:11.1
+pin_run_as_build:
+  python:
+    min_pin: x.x
+    max_pin: x.x
+python:
+- 3.10.* *_cpython
+pytorch:
+- 1.13.0
```
Comment (h-vetinari) [inline .ci_support/linux_64_c_compiler_version10cuda_compiler_version11.1cxx_compiler_version10python3.10.____cpythonpytorch1.13.0.yaml:28]: You can see the actual string being used in the config yamls.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -13,10 +13,13 @@ source:
...
+  # as of pytorch 1.13, conda-forge only builds for CUDA 11.2+, see
+  # https://github.com/conda-forge/conda-forge-pinning-feedstock/issues/3491
+  skip: true  # [pytorch >= "1.13" and cuda_compiler_version in ("10.2", "11.0", "11.1")]
```
Comment (h-vetinari) [inline recipe/meta.yaml:22]: Just need to do exact matching (or: stop building for multiple pytorch versions, which shouldn't be necessary given the migration infrastructure)

```suggestion
  skip: true  # [pytorch == "1.13.0" and cuda_compiler_version in ("10.2", "11.0", "11.1")]
```

**Comment by peastman** [comment]
That worked, thanks!  @mikemhenry I'm going to merge this so we can see whether it fixes the installation problems.

**Pull request merged**